### PR TITLE
Fixes multi line textfield hint text gets ellipsized

### LINKED ISSUE #148353: [A11y] TextField should adapt to hint text sizes
b/317131589

### Steps to reproduce

run app and activate the textfield

### Expected results

According to internal guideline the the hint text can't be ellipisized when font size increases. 

The material guideline suggest no ellipsis and no multiline
https://m3.material.io/components/text-fields/guidelines.

The native android expands the text field size to accommodate long string
image

Since we can't control how long the text developer can put into the hint text, I think we should follow native android to make it multi line and expand the textfield size.

### Actual results

image

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: HomePage(),
  ));
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text("Home")),
      body: Center(
        child: Column(
          children: [
            TextField(
              decoration: const InputDecoration(
                labelText: 'Email',
                suffixText: '@gmail.com',
                hintText: 'Enter your email.... Please, I really need it',
              ),
              controller: TextEditingController(text: 'abc'),
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console

```

</details>

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/148353

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM but watch out for Google test failures.

Also see this old issue: https://github.com/flutter/flutter/issues/46182. It might be nicer to keep the ellipsis if needed but expand to maxLines as suggested there, but I'm not sure how we would implement that exactly. Since this affects accessibility, it probably makes sense to just merge this fix and migrate any broken Google tests.

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -1789,6 +1789,27 @@ void main() {
...
+          labelText: 'Email',
+          hintText: kLongString,
+        ),
+      ),
+    ));
+    final Text hintText = tester.widget<Text>(find.text(kLongString));
+    expect(hintText.overflow, isNull);
+    final RenderParagraph paragraph = tester.renderObject<RenderParagraph>(find.text(kLongString));
+    expect(paragraph.size.height > defaultLineHeight * 2, isTrue);
+  }); // [intended] only applies to platforms where we supply the context menu.
```
Comment (justinmc) [inline packages/flutter/test/material/text_field_test.dart:1811]: Nit: Is it worth it to add another test where maxLines is 2 and kLongString is long enough to still be truncated?

**Comment by justinmc** [comment]
@chunhtai FYI there are still some failures here.

**Comment by chunhtai** [comment]
some update: still looking at google test failure

**Pull request merged**